### PR TITLE
enhance: reduce 1x memory copy when loading json

### PR DESCRIPTION
**Pull request by longjiquan**
/kind improvement

### CODE REVIEW

**Comment by czs007** [comment]
@longjiquan need pick to 2.3 branch

File: internal/core/src/common/FieldDataInterface.h [hunk 1]
```diff
@@ -290,6 +290,14 @@ class FieldDataJsonImpl : public FieldDataImpl<Json, true> {
         return field_data_[offset].data().size();
     }
 
+    void
+    FillFieldData(const std::shared_ptr<arrow::Array> array) override {
+        AssertInfo(array->type()->id() == arrow::Type::type::BINARY,
+                   "inconsistent data type");
```
Comment (yah01) [inline internal/core/src/common/FieldDataInterface.h:296]: add the `array->type()` in the error message

**Comment by xiaofan-luan** [comment]
/lgtm
/approve

good catch

**Comment by czs007** [comment]
manual merge

**Pull request merged**